python/GafferUI/SpreadsheetUI.py

```diff
@@ -2083,53 +2083,40 @@ def __prependSpreadsheetCreationMenuItems( menuDefinition, plugValueWidget ) :
 	if plug.getInput() is not None or not plugValueWidget._editable() or Gaffer.MetadataAlgo.readOnly( plug ) :
 		return
 
-	ancestorPlug, ancestorLabel = None, ""
-	for ancestorType, ancestorLabel in [
-		( Gaffer.TransformPlug, "Transform" ),
-		( Gaffer.Transform2DPlug, "Transform" ),
-		( Gaffer.NameValuePlug, "Value and Switch" ),
-	] :
-		ancestorPlug = plug.ancestor( ancestorType )
-		if ancestorPlug :
-			if all( p.getInput() is None for p in Gaffer.Plug.RecursiveRange( ancestorPlug ) ) :
-				break
-			else :
-				ancestorPlug = None
+	plugsAndSuffixes = [ ( plug, "" ) ]
+
+	ancestorPlug = plug.parent()
+	while isinstance( ancestorPlug, Gaffer.Plug ) :
+
+		if any( p.getInput() is not None for p in Gaffer.Plug.RecursiveRange( ancestorPlug ) ) :
+			break
+
+		if isinstance( ancestorPlug, ( Gaffer.TransformPlug, Gaffer.Transform2DPlug ) ) :
+			plugsAndSuffixes.append( ( ancestorPlug, " (Transform)" ) )
```

Comment:
Sorry, silly Q - why are Transform plugs special cased here?

Reply:
Moved to a standard metadata registration in d120f0a.